java/flight/src/main/java/org/apache/arrow/flight/Location.java

```diff
@@ -17,34 +17,51 @@
 
 package org.apache.arrow.flight;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+
 import org.apache.arrow.flight.impl.Flight;
 
+/** A URI where a Flight stream is available. */
 public class Location {
+  private final URI uri;
 
-  private final String host;
-  private final int port;
-
-  public Location(String host, int port) {
+  public Location(String uri) throws URISyntaxException {
```

Comment:
if this is public, we should validate the provided uri to make sure it is supported.